include/pistache/endpoint.h

```diff
@@ -49,6 +49,45 @@ class Endpoint {
 
     void shutdown();
 
+    /*!
+     * \brief Use SSL on this endpoint
+     *
+     * \param[in] cert Server certificate path
+     * \param[in] key Server key path
+     * \param[in] use_compression Wether or not use compression on the encryption
+     *
+     * Setup the SSL configuration for an endpoint. In order to do that, this
+     * function will init OpenSSL constants and load *all* algorithms. It will
+     * then load the server certificate and key, in order to use it later.
+     * *If the private key does not match the certificate, an exception will
+     * be thrown*
+     *
+     * \note use_compression is false by default to mitigate BREACH[1] and
+     *          CRIME[2] vulnerabilities
+     * \note This function will throw an exception if pistache has not been
+     *          compiled with PISTACHE_USE_SSL
+     *
+     * [1] https://en.wikipedia.org/wiki/BREACH
+     * [2] https://en.wikipedia.org/wiki/CRIME
+     */
+    void useSSL(std::string cert, std::string key, bool use_compression = false);
```

Comment:
IMO from the performance perspective it's may be better to pass both, cert and key as 
constant references (const std::string&) rather than copies of the strings (std::string)